app/models/game.rb

```diff
@@ -30,27 +39,44 @@ def populate_game!
     save
   end
 
-  def check?
-    white_king = pieces.find_by(type: 'King', color: 'white')
-    black_pieces = pieces.where(color: 'black')
+  # Keep track of current king.  We don't care what the other king is doing.
+  def check?(color)
+    king = pieces.find_by(type: 'King', color: color)
+    opposing_pieces = pieces.where(color: king.opposite_color, captured: false)
```

Comment:
Added captured: false 
Otherwise, dead pieces get looked at and mess up valid move